src/Compilers/CSharp/Test/Emit/Emit/EntryPointTests.cs

```diff
@@ -1642,5 +1642,19 @@ class Program
                 // error CS5001: Program does not contain a static 'Main' method suitable for an entry point
                 Diagnostic(ErrorCode.ERR_NoEntryPoint));
         }
+
+        [Fact]
+        public void AbstractEntryPoint()
+        {
+            var source = @"
+interface Program
+{
+    static abstract void Main();
```

Comment:
Let's also include a positive test when you implement the `Program` interface and run it (have the impl print 1 or something trivial). To do this, you'll want to call `CompileAndVerify`, and then use `ExecutionConditionUtil.IsCoreClr` to pass in either the expected output or null (to skip running the binary) based on whether the test is executing on a framework that can run static abstracts.